pkg/auto/lights/auto.go

```diff
@@ -181,25 +185,32 @@ func (b *BrightnessAutomation) processStateChanges(ctx context.Context, readStat
 			}
 
 			// if the context remains live, schedule another update soon
-			// TODO: make this delay configurable, or add backoff etc.
-			after := 5 * time.Second
+			b.retries++
+			after := time.Duration(b.retries) * b.backoffMultiplier
 			b.logger.Error("processState failed; scheduling retry",
 				zap.Error(err),
 				zap.Duration("retryAfter", after),
 			)
 			ttl = after
 		}
 
-		if ttl < 0 {
-			b.logger.Warn("ttl < 0; using refreshEvery instead")
+		if err == nil || b.retries > b.maxRetries {
+			b.retries = 0
 		}
+
 		// ensure it's not too long before we wake up, so the lights are refreshed regularly
 		// so external changes don't stick around forever
-		if ttl <= 0 || ttl > refreshEvery {
+		if ttl < 0 {
+			b.logger.Warn("ttl < 0; using refreshEvery instead")
+			ttl = b.refreshEvery.Load()
+		}
+		if ttl > b.refreshEvery.Load() {
+			b.logger.Warn("ttl > refreshEvery; using refreshEvery instead")
 			// b.logger.Debug("waking up sooner to ensure lights aren't stale",
 			// 	zap.Duration("after", refreshEvery))
-			ttl = refreshEvery
+			ttl = b.refreshEvery.Load()
 		}
```

Comment:
Careful here - The first `b.refreshEvery.Load()` could return a different value to the second one, causing an unintended value to be stored in `ttl`. Load the value only once.